prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/git/GitArchiveTest.java

```diff
@@ -0,0 +1,124 @@
+package com.redhat.parodos.tasks.git;
+
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import junit.framework.TestCase;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class GitArchiveTest extends TestCase {
+
+	private GitArchive gitArchive;
+
+	private Path gitRepoPath;
+
+	private Repository repository;
+
+	private Path tempDir;
+
+	@BeforeEach
+	public void setUp() throws Exception {
+		gitArchive = new GitArchive(new GitUtils());
+		tempDir = Files.createTempDirectory("git-repo");
+
+		gitRepoPath = tempDir.resolve(".git");
+
+		this.repository = FileRepositoryBuilder.create(gitRepoPath.toFile());
+		this.repository.create();
+		System.out.println("Created a new repository at " + this.repository.getDirectory());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		repository.close();
+		Files.walk(tempDir).sorted(java.util.Comparator.reverseOrder()).map(Path::toFile).forEach(p -> {
```

Comment:
`Files.walk` returns a stream, please use inside a try with resources.

Reply:
Not sure if I follow this